### DISCUSSION TITLE
Ability to generate TypeSpec API specs

### DISCUSSION THREAD

**Discussion by michaelgmiller1** [fastapi/fastapi#12237] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
n/a
```

### Description

Microsoft released a great tool called TypeSpec: https://typespec.io/. It would be great if FastAPI could auto-generate specifications in TypeSpec format. This would let us have our frontend share types with our backend easily with full toolchain integration.

### Operating System

macOS

### FastAPI Version

n/a

### Pydantic Version

n/a

### Python Version

n/a

**Comment by Kludex** [comment]
Hi @michaelgmiller1 ,

How generating the TypeSpec file would help your frontend? Wouldn't you still need to generate the OpenAPI, and then the frontend client?

**Reply by michaelgmiller1** [reply]
So we don't currently type our frontend client from backend code.. but if we did have a TypeSpec file, we could likely import it directly into our Frontend, with no need to generate OpenAPI code.

**Reply by Kludex** [reply]
Can you import `.tsp` from the frontend?

**Reply by michaelgmiller1** [reply]
Not directly quite yet. They are working on a TypeScript emitter (and have one for testing purposes): https://github.com/microsoft/typespec/discussions/3307.. I would be shocked if there wasn't an importer plugin for `tsp` files in the near future, given the almost 1:1 parity from `tsp` to `ts`.
